### PR TITLE
Use Rubocop v0.36.0

### PR DESCRIPTION
**Pull request by nickhoffman**
## Problem

Policial uses Rubocop v0.35.1 , which is missing bug fixes.
## Solution

Upgrade Rubocop to the current version, which is v0.36.0 .

/review @volmer, @etiennebarrie
/cc @mariusbutuc

### CODE REVIEW

**Comment by nickhoffman** [comment]
Note: Some specs are failing, but I haven't been able to figure out why. They all relate to `Rubocop::Cop::Cop` objects having a `nil` value in `@all` . E.g.

```
  1) Policial::StyleChecker#violations skips style guides on files that they are not able to investigate
     Failure/Error: cop_classes.reject!(&:non_rails) unless config['AllCops'].fetch('Rails', {})['Enabled']

     NoMethodError:
       undefined method `without_type' for nil:NilClass
     # ./vendor/bundle/gems/rubocop-0.36.0/lib/rubocop/cop/cop.rb:82:in `all'
     # ./vendor/bundle/gems/rubocop-0.36.0/lib/rubocop/cop/cop.rb:90:in `non_rails'
     # ./lib/policial/style_guides/ruby.rb:31:in `reject!'
     # ./lib/policial/style_guides/ruby.rb:31:in `team'
     # ./lib/policial/style_guides/ruby.rb:10:in `violations_in_file'
     # ./lib/policial/style_checker.rb:21:in `block (2 levels) in violations_in_checked_files'
     # ./lib/policial/style_checker.rb:19:in `each'
     # ./lib/policial/style_checker.rb:19:in `flat_map'
     # ./lib/policial/style_checker.rb:19:in `block in violations_in_checked_files'
     # ./lib/policial/style_checker.rb:18:in `each'
     # ./lib/policial/style_checker.rb:18:in `flat_map'
     # ./lib/policial/style_checker.rb:18:in `violations_in_checked_files'
     # ./lib/policial/style_checker.rb:12:in `violations'
     # ./spec/style_checker_spec.rb:58:in `block (3 levels) in <top (required)>'
```

**Comment by volmer** [comment]
Wow, RuboCop 0.36 is huge! https://github.com/bbatsov/rubocop/releases/tag/v0.36.0

I'll check the changelog to understand exactly what's behind those CI failures

File: lib/policial/style_guides/ruby.rb [hunk 1]
```diff
@@ -28,7 +28,7 @@ def default_config_file
 
       def team
         cop_classes = RuboCop::Cop::Cop.all
-        cop_classes.reject!(&:rails?) unless config['AllCops']['RunRailsCops']
+        cop_classes.reject!(&:non_rails) unless config['AllCops'].fetch('Rails', {})['Enabled']
```
Comment (volmer) [inline lib/policial/style_guides/ruby.rb]: [According to the docs](https://github.com/bbatsov/rubocop#rails) `Rails` does not go under `AllCops`.

Comment (volmer) [inline lib/policial/style_guides/ruby.rb]: What about

``` ruby
        cop_classes =
          if config['Rails']['Enabled']
            RuboCop::Cop::Cop.all
          else
            RuboCop::Cop::Cop.non_rails
          end
```

**Comment by volmer** [comment]
`RuboCop::ProcessedSource` now requires the target Ruby version: https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/processed_source.rb#L23

You can get the version on [`ruby.rb`](https://github.com/volmer/policial/blob/master/lib/policial/style_guides/ruby.rb) using

``` ruby
config['AllCops']['TargetRubyVersion']
```

**Comment by nickhoffman** [comment]
Thanks for the tips, @volmer. Yeah, Rubocop v0.36.0 is massive.

The target Ruby version is being given to `RuboCop::ProcessedSource` now, and the repo targets Ruby 2.1, and a few new Rubocop violations have been fixed. Could you review this again, please?

**Comment by volmer** [comment]
I like it! :+1: 
Just one last thing: could you please squash your commits?

**Comment by nickhoffman** [comment]
Sure, I can squash them. Out of curiousity, why? Isn't it better to preserve the steps that were needed to implement this?

**Comment by volmer** [comment]
Commits should be atomic enough to be reverted without causing inconsistencies on the repo history or broken builds (ideally). Also, the fewer commits we have, the better is the `git log` (good messages are equally important). Sure, in PRs with bigger changes multiple commits might make sense, but on this one I don't see a reason to have more than one commit.

**Comment by nickhoffman** [comment]
Fair enough! The commits've been squashed.

**Comment by volmer** [comment]
Thank you! :100:

**Comment by nickhoffman** [comment]
And thank _you_ for building and maintaining Policial!

**Comment by nickhoffman** [comment]
Whoops, I just realized that we probably should've bumped the minor version in this gem. Shall I submit a PR for that, and then a PR in Shopify/policial.io to bump the version of Policial?

**Comment by volmer** [comment]
Don't worry, Policial.io is updated based on the latest `master`. I'll do it soon.

**Pull request merged**